Bottom: 7f5cac7ae2f22d13595f044007c96f8c1f272f39
Top:    cd88d9fdde844f5987cce753bd199b3bff89367c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-11 16:35:01 +0000

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 65a1958..6dc8b76 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -708,7 +708,7 @@ lookup_seg (PLINE *contour, VNODE *vertex)
   /* fill in the segment in info corresponding to this node */
   if (setjmp (info.sego) == 0)
     {
-      r_search_1 (contour->tree, &box, NULL, get_seg, &info);
+      r_search (contour->tree, &box, NULL, get_seg, &info);
       assert (0);
     }
   return info.s;
@@ -1145,7 +1145,11 @@ label_contour (PLINE * a)
           continue;
         }
     }
-  while ((cur = cur->next) != first_not_unknown);
+  while ((cur = cur->next) != first_not_unknown &&
+         !(cur->next == &a->head && first_not_unknown == NULL));
+
+  if (first_not_unknown == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
